### ISSUE TITLE
io.popen: Windows path behavior

### ISSUE THREAD

**Issue by NicolaiSoeborg** [yuin/gopher-lua#90]
1. What version of GopherLua are you using? : Newest (commit 7692488a1ad6bd06dc48890f4a149b65a86a767d)

2. What version of Go are you using? : `go version go1.7.4 linux/amd64`

3. What operating system and processor architecture are you using? : Windows Subsystem for Linux / Ubuntu (`Linux 4.4.0-43-Microsoft #1-Microsoft Wed Dec 31 14:42:53 PST 2014 x86_64 x86_64 x86_64 GNU/Linux`)

4. What did you do? :
Running the following code:
```
package main
import "fmt"
import (
    "github.com/yuin/gopher-lua"
)

func main() {
        L := lua.NewState()
        defer L.Close()

        fmt.Println(`Testing: IF EXIST "C:/someFile/*" (ECHO a) ELSE (ECHO b)`)
        if err := L.DoString(`print(io.popen('IF EXIST "C:/someFile/*" (ECHO a) ELSE (ECHO b)'):read(1))`); err != nil {
                panic(err)
        }
        fmt.Println("")
        fmt.Println(`Testing: IF EXIST "C:/someDir/*" (ECHO a) ELSE (ECHO b)`)
        if err := L.DoString(`print(io.popen('IF EXIST "C:/someDir/*" (ECHO a) ELSE (ECHO b)'):read(1))`); err != nil {
                panic(err)
        }
}
```
Compiled using `GOOS=windows GOARCH=386 go build file.go`
Running on a windows system (`Windows 10, build 14986.1000`) with "C:\\someFile" being a file and "C:\\someDir" being a directory.

5. What did you expect to see? :
```
Testing: IF EXIST "C:/someFile/*" (ECHO a) ELSE (ECHO b)
b

Testing: IF EXIST "C:/someDir/*" (ECHO a) ELSE (ECHO b)
a

```

6. What did you see instead? :
```
Testing: IF EXIST "C:/someFile/*" (ECHO a) ELSE (ECHO b)
b

Testing: IF EXIST "C:/someDir/*" (ECHO a) ELSE (ECHO b)
b

```

Running the commands manually from `cmd` gives the expected result: http://i.imgur.com/AtNZNOc.png

I can most likely use LFS to avoid this "see if path is dir"-hack, but just found the different behaviors odd.

**Comment by yuin** [comment]
The original C Lua implementation uses [_popen](https://msdn.microsoft.com/en-us//library/96ayss4b.aspx) to create a new process. But Go does not have an exactly equivalent library function for it as far as I know.  

So, GopherLua uses 'exec.Command' function for emulating _popen's behaviour.

>> Running the commands manually from cmd gives the expected result: http://i.imgur.com/AtNZNOc.png

Therefore, your confirmation process is somewhat wrong.

`cmd.exe /c "IF EXIST \"C:/someDir/*\" (ECHO a) ELSE (ECHO b)"` is the command that is actually spawned by `io.popen` in GopherLua. This command shall outputs `b` .

I think that this behaviour is not avoidable as far as using Go. Do you know any Go function equivalent to `_popen`?.

**Comment by NicolaiSoeborg** [comment]
> I think that this behaviour is not avoidable as far as using Go. Do you know any Go function equivalent to `_popen`?.

I'm just getting to know go, so nope! (however, I have been impressed by how easy it is to integrate gopher-lua 👍 )


After some more testing, it seems like Windows path handling is very _sensitive_:
![testing dirs in windows](https://i.imgur.com/07A4cUK.png)


I managed to solve my problem (https://github.com/NicolaiSoeborg/filetree-plugin/commit/0d3f484c64f8adc314c24045cc3d10cf78f2f207) by simply not escaping the path! 😮 

I do not think this weird behavior is for gopher-lua to solve, so I've closed the ticket.
